src/main/java/frc/robot/commands/ControlledHeadingDriveCommand.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants.OIConstants;
+import frc.robot.subsystems.DrivetrainSubsystem;
+
+public class ControlledHeadingDriveCommand extends CommandBase {
+  /** Creates a new ControlledHeadingDriveCommand. */
+    private final XboxController m_controller;
+    private final DrivetrainSubsystem m_drivetrainSubsystem;
+    
+    public ControlledHeadingDriveCommand(XboxController m_controller, DrivetrainSubsystem m_drivetrainSubsystem) {
```

Comment:
``m_`` should not be used in parameters or variables that are not class-level! Plus, look at line 15, we don't want to reuse variable names, even if Java does show an error.